contracts/impl/TaskImpl.sol

```diff
@@ -25,18 +25,19 @@ library TaskImpl {
     using Bytes for uint64;
     using Bytes for address;
 
-    event TaskRecordCreated(bytes32 taskId, bytes32 inputsHash, uint64 gasLimit, uint64 gasPx, address sender,
+    event TaskRecordCreated(bytes32 indexed taskId, bytes32 inputsHash, uint64 gasLimit, uint64 gasPx, address sender,
         uint blockNumber);
     // SecretContractDeployed => bytes32s [taskId, preCodeHash, codeHash, initStateDeltaHash]
-    event SecretContractDeployed(uint64 gasUsed, address optionalEthereumContractAddress, bytes32[4] bytes32s,
-        uint gasUsedTotal, bytes optionalEthereumData, address workerAddress);
+    event SecretContractDeployed(bytes32 indexed taskId, uint64 gasUsed, address optionalEthereumContractAddress,
+        bytes32[4] bytes32s, uint gasUsedTotal, bytes optionalEthereumData, address workerAddress);
     // ReceiptVerified => bytes32s [scAddr, taskId, stateDeltaHash, outputHash]
-    event ReceiptVerified(uint64 gasUsed, address optionalEthereumContractAddress, bytes32[4] bytes32s,
-        uint deltaHashIndex, uint gasUsedTotal, bytes optionalEthereumData, address workerAddress, bytes sig);
-    event ReceiptFailed(bytes32 taskId, bytes32 scAddr, uint gasUsed, address workerAddress, bytes sig);
-    event ReceiptFailedETH(bytes32 taskId, bytes32 scAddr, uint gasUsed, uint gasUsedTotal, address workerAddress,
-        bytes sig);
-    event TaskFeeReturned(bytes32 taskId);
+    event ReceiptVerified(bytes32 indexed taskId, uint64 gasUsed, address optionalEthereumContractAddress,
+        bytes32[4] bytes32s, uint deltaHashIndex, uint gasUsedTotal, bytes optionalEthereumData, bytes sig,
+        address workerAddress);
+    event ReceiptFailed(bytes32 indexed taskId, bytes32 scAddr, uint gasUsed, address workerAddress, bytes sig);
+    event ReceiptFailedETH(bytes32 indexed taskId, bytes32 scAddr, uint gasUsed, uint gasUsedTotal,
```

Comment:
This is great!
https://github.com/enigmampc/enigma-p2p/blob/3069a5cdf0d81ed72e9a0307c9c5188a1426072a/src/ethereum/EnigmaContractProductionWriterAPI.js#L143-L147